File: app/db/models/SentenceTranslation.ts
import {
  Model,
  Column,
  Table,
  ForeignKey,
  HasMany,
  BelongsToMany,
  BelongsTo,
  DefaultScope,
  HasOne,
} from 'sequelize-typescript';
import Sentence from './Sentence';

@Table
class SentenceTranslation extends Model {
  @ForeignKey(() => Sentence)
  @BelongsTo(() => Sentence, { targetKey: 'tatoebaId', as: 'translation' })
  @Column
  sentenceId!: number;

  @Column
  translationId!: number;
}

export default SentenceTranslation;

File: rails-backend/app/models/sentence_translation.rb
class SentenceTranslation < ApplicationRecord
  belongs_to :sentence, foreign_key: 'sentenceId', primary_key: 'tatoebaId'
  belongs_to :translation, class_name: 'Sentence', foreign_key: 'translationId', primary_key: 'tatoebaId'

  # # this one 
  # has_one :translation, class_name: 'Sentence', foreign_key: 'tatoebaId', primary_key: 'translationId'
end
